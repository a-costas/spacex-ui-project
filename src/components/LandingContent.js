import React from 'react';
import { gql, useQuery } from '@apollo/client';
import Logo from '../resources/spacex-brand.png';

const GET_COMPANY_INFO = gql`
    {
        company {
            name
            founded
            summary
            valuation
            employees
            founder
            headquarters {
                city
                state
            }
        }
    }
`

const LandingContent = () => {

    const { data, loading, error } = useQuery(GET_COMPANY_INFO);

    if (loading) return (
        <p>'Loading...'</p>
    );

    if (error) return (
        <p>`Error! ${error.name}: ${error.message}`</p>
    );

    return (
        <div className='landing-content'>
            <div className="company-name">
                <img className="landing-brand" src={Logo} alt="spacex-logo" />
            </div>
            <div className="company-about">
                { data.company.name } was founded in { data.company.founded } by { data.company.founder }. { data.company.summary } Today, { data.company.name } employs { data.company.employees } and 
                is valuated at { data.company.valuation } dollars. Its headquarters are in { data.company.headquarters.city }, { data.company.headquarters.state }.
            </div>
        </div>
    );
}

export default LandingContent;
export {GET_COMPANY_INFO};