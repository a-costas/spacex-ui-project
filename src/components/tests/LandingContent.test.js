import { waitFor } from '@testing-library/react';
import LandingContent from '../LandingContent';
import { GET_COMPANY_INFO } from '../LandingContent';
import TestRenderer from 'react-test-renderer';
import { MockedProvider } from '@apollo/client/testing';

const mocks = [
    {
        request: {
            query: GET_COMPANY_INFO,
            variables: {},
        },
        result: {
            data: {
                company: {
                    name: 'SpaceX',
                    founded: 2002,
                    summary: 'SpaceX designs, manufactures and launches advanced rockets and spacecraft. The company was founded in 2002 to revolutionize space technology, with the ultimate goal of enabling people to live on other planets.',
                    valuation: 27500000000,
                    employees: 7000,
                    founder: 'Elon Musk',
                    headquarters: {
                        city: 'Hawthorne',
                        state: 'California'
                    }
                }
            }
        }
    }
];

it('renders without error', () => {
    const component = TestRenderer.create(
        <MockedProvider mocks={mocks} addTypename={false}>
            <LandingContent />
        </MockedProvider>
    );

    const tree = component.toJSON();
    expect(tree.children[0]).toContain('Loading...');
})

it('should render landing content', () => {
    const component = TestRenderer.create(
        <MockedProvider mocks={mocks} addTypename={false}>
            <LandingContent />
        </MockedProvider>
    );

    expect(component).toMatchSnapshot();
});

it('should render landing content', async () => {
    const component = TestRenderer.create(
        <MockedProvider mocks={mocks} addTypename={false}>
            <LandingContent />
        </MockedProvider>
    );

    await waitFor(() => new Promise((res) => setTimeout(res, 0)));

    expect(component).toMatchSnapshot();
})