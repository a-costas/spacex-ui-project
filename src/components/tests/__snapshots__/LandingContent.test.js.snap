// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render landing content 1`] = `
<p>
  'Loading...'
</p>
`;

exports[`should render landing content 2`] = `
<p>
  'Loading...'
</p>
`;
